src/components/simulator/Simulator.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styles from './Simulator.module.css';
+import { ReactComponent as SimulatorIcon } from '../../assets/images/icons/Simulator.svg';
+import AttachFileIcon from '@material-ui/icons/AttachFile';
+import { Button } from '@material-ui/core';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import DefaultWhatsappImage from '../../assets/images/whatsappDefault.jpg';
+import { useQuery } from '@apollo/client';
+import Draggable from 'react-draggable';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import InsertEmoticonIcon from '@material-ui/icons/InsertEmoticon';
+import MicIcon from '@material-ui/icons/Mic';
+import CallIcon from '@material-ui/icons/Call';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import VideocamIcon from '@material-ui/icons/Videocam';
+import CameraAltIcon from '@material-ui/icons/CameraAlt';
+import Axios from 'axios';
+import { SEARCH_QUERY } from '../../graphql/queries/Search';
+import { SEARCH_QUERY_VARIABLES } from '../../common/constants';
+import moment from 'moment';
+import { TIME_FORMAT } from '../../common/constants';
+import ClearIcon from '@material-ui/icons/Clear';
+import { GUPSHUP_CALLBACK_URL, SIMULATOR_CONTACT } from '../../config';
+
+export interface SimulatorProps {
+  showSimulator: boolean;
+  setShowSimulator: any;
+}
+
+export const Simulator: React.FC<SimulatorProps> = ({
+  showSimulator,
+  setShowSimulator,
+}: SimulatorProps) => {
+  const [inputMessage, setInputMessage] = useState('');
+  const messageRef: any = useRef<HTMLDivElement>();
+
+  let messages = [];
+  let simulatorId = '';
+
+  const { data: allConversations }: any = useQuery(SEARCH_QUERY, {
```

Comment:
maybe here you should get only simulator contact conversations.

Reply:
We are filtering the contact from all conversations query because when we send or receive a message from subscription, only this query is updated and the changes are reflected on the screen. If we will not call this query and get only the messages of the particular contact it will be difficult to update on sending and receiving messages in the simulator.

Reply:
Ok make sense.